rdt-open-reader/src/testCovidIndonesia/java/io/ona/rdt/robolectric/activity/CovidPatientRegisterActivityTest.java

```diff
@@ -72,15 +76,18 @@ public void testLanguageSwitcherDialogShouldSetCorrectLocale() throws Exception
 
     private void verifyLocaleIsSaved(ListView listView, int position, String locale) {
         listView.performItemClick(null, position, listView.getItemIdAtPosition(position));
-        Assert.assertEquals(locale, LangUtils.getLanguage(RDTApplication.getInstance()));
-        Assert.assertEquals(locale, RDTApplication.getInstance().getResources().getConfiguration().locale.getLanguage());
+        Assert.assertEquals(locale, new Locale(RDTApplication.getInstance().getSharedPreferences().fetchLanguagePreference()).getLanguage());
+        Assert.assertEquals(locale, covidPatientRegisterActivity.getResources().getConfiguration().locale.getLanguage());
     }
 
     @Test
     public void testRegisterLanguageSwitcherShouldVerifyLocaleIndex() throws Exception {
-        verifySavedLanguageIndex(1);
-        LangUtils.saveLanguage(covidPatientRegisterActivity, locales[0]);
         verifySavedLanguageIndex(0);
+        Resources resources = covidPatientRegisterActivity.getResources();
+        Configuration configuration = resources.getConfiguration();
+        configuration.setLocale(new Locale(locales[1]));
+        resources.updateConfiguration(configuration, resources.getDisplayMetrics());
+        verifySavedLanguageIndex(1);
```

Comment:
Not sure I understand what this is testing? Is this meant to verify that the default language is selected?